### DISCUSSION TITLE
Helpful features for xdg-desktop-portal-termfilechooser

### DISCUSSION THREAD

**Discussion by poperigby** [sxyazi/yazi#1136] [4. Ideas]
I'm implementing a script for [xdg-desktop-portal-termfilechooser](https://github.com/GermainZ/xdg-desktop-portal-termfilechooser), and there are a few missing features from Yazi that would make the experience better.

* Ability to select directories with `--chooser-file`: This would be helpful for when the portal requests that you choose a directory. I'm currently using `--cwd-file`, entering a directory, and then quitting but the UX would be better if you could select a directory.
* Ability to show a message using a command line flag: This would be helpful for giving some context. 
* Flag for only showing directories: For when the portal asks for us to choose a directory

If these could be implemented with a plugin easier, that would be fine too.

Thanks for making Yazi! :slightly_smiling_face:

**Comment by exquo** [comment]
👍 on the suggestions. Yazi is convenient to use with `termfilechooser` as it is fast to start up.

Regarding the first one: `--chooser-file` should work with directories the same way as with the regular files. Select a directory with `o` or `Enter` (unless you have configured yazi's [Smart enter](https://yazi-rs.github.io/docs/tips#smart-enter)). I have added a [PR](https://github.com/boydaihungst/xdg-desktop-portal-termfilechooser/pull/4/files#diff-a2f2a12c232f9c8ddf2bd5df318d3df193fc48c4e664203a116c43a5fbf146f9R24-R25) for yazi wrapper that combines `--chooser-file` and `--cwd-file` options for directories selection.
